### PR TITLE
Update typescript typings to be stricter

### PR DESCRIPTION
**Pull request by luke-john**
**What**:
Made the typescript definitions stricter so when adding css props to a html element the compiler will complain on misspellings and incorrect usage of props.

**Why**:
Because it gives confidence that styles are used correctly and gives you hints about what properties are valid.

image

**How**:
Using the Definitions from react around what are valid style properties for HTML and SVG elements.

### CODE REVIEW

**Comment by kwelch** [comment]
This might be a good PR to take care of #50, if you understand how.

File: typings/glamorous.d.ts [hunk 1]
```diff
@@ -3,137 +3,33 @@
 // Definitions by: Kok Sam <https://github.com/sammkj>
 
 import * as React from "react";
-import { StatelessComponent, ComponentClass, PureComponent } from "react";
+import { ComponentClass, CSSProperties, StatelessComponent, SVGAttributes } from 'react'
```
Comment (sammkj) [inline typings/glamorous.d.ts]: Can you change single quote to double quote?

Comment (kentcdodds) [inline typings/glamorous.d.ts]: Actually, we should follow the standard of the project. Could you use single quotes only please?

Comment (sammkj) [inline typings/glamorous.d.ts]: Ahh sorry, I didn't notice!

Comment (luke-john) [inline typings/glamorous.d.ts]: updated initial react import to use singles as well.

**Comment by luke-john** [comment]
Because I've split the typings into multiple files this pr would be broken by https://github.com/paypal/glamorous/pull/51.

A couple of possible options are
1. I can update this pr to have the definitions in a single file.
2. #51 could be updated to copy the `.d.ts` files to a typings folder under dist.
3. Hold off on landing this pr until 51 lands and I update this pr to include the solution in option 2.

**Comment by kentcdodds** [comment]
Thanks @luke-john! I think we should probably just publish the `typings` directory instead. Is there any way we could move everything except `.d.ts` files out of that directory? We could add a `test` directory for the TypeScript test file. Does that sound ok? If you could update the `files` property in the `package.json` to include `typings`, and make those other changes, I think we can close #54 and just use this one :+1:

**Comment by luke-john** [comment]
I've updated the structure as following in an additional commit

```
/test
  - glamorous.test.ts
/typings
  - ...definition files
- tsconfig.json
- tslint.json
```

**Comment by kentcdodds** [comment]
I'd like for @sammkj to review the changes, then we can merge it :)

**Comment by sammkj** [comment]
@kentcdodds LGTM! Let's merge this

**Pull request merged**